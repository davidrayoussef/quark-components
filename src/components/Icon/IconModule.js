import React from 'react';
import ArrowLeftIcon from './icons/ArrowLeftIcon';
import ArrowRightIcon from './icons/ArrowRightIcon';
import ArrowDownIcon from './icons/ArrowDownIcon';
import CloseIcon from './icons/CloseIcon';
import EmailIcon from './icons/EmailIcon';
import FacebookIcon from './icons/FacebookIcon';
import GithubIcon from './icons/GithubIcon';
import MenuIcon from './icons/MenuIcon';
import RedditIcon from './icons/RedditIcon';
import TwitterIcon from './icons/TwitterIcon';
import YoutubeIcon from './icons/YoutubeIcon';

const icons = {
  arrowLeft: <ArrowLeftIcon />,
  arrowRight: <ArrowRightIcon />,
  arrowDown: <ArrowDownIcon />,
  close: <CloseIcon />,
  email: <EmailIcon />,
  facebook: <FacebookIcon />,
  github: <GithubIcon />,
  menu: <MenuIcon />,
  reddit: <RedditIcon />,
  twitter: <TwitterIcon />,
  youtube: <YoutubeIcon />
};

export default icons;
