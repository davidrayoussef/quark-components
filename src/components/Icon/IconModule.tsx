import React from 'react';

import {
  ArrowLeftIcon,
  ArrowRightIcon,
  ArrowDownIcon,
  CloseIcon,
  EmailIcon,
  FacebookIcon,
  GithubIcon,
  MenuIcon,
  RedditIcon,
  TwitterIcon,
  YoutubeIcon
} from './icons';

export const icons: { [key: string]: JSX.Element } = {
  arrowLeft: <ArrowLeftIcon />,
  arrowRight: <ArrowRightIcon />,
  arrowDown: <ArrowDownIcon />,
  close: <CloseIcon />,
  email: <EmailIcon />,
  facebook: <FacebookIcon />,
  github: <GithubIcon />,
  menu: <MenuIcon />,
  reddit: <RedditIcon />,
  twitter: <TwitterIcon />,
  youtube: <YoutubeIcon />
};
