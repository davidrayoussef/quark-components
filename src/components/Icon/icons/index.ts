export { ArrowDownIcon } from './ArrowDownIcon';
export { ArrowLeftIcon } from './ArrowLeftIcon';
export { ArrowRightIcon } from './ArrowRightIcon';
export { CloseIcon } from './CloseIcon';
export { EmailIcon } from './EmailIcon';
export { FacebookIcon } from './FacebookIcon';
export { GithubIcon } from './GithubIcon';
export { MenuIcon } from './MenuIcon';
export { MinusIcon } from './MinusIcon';
export { PlusIcon } from './PlusIcon';
export { RedditIcon } from './RedditIcon';
export { TwitterIcon } from './TwitterIcon';
export { YoutubeIcon } from './YoutubeIcon';
