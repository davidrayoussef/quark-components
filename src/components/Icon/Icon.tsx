import React from 'react';

import { icons } from './IconModule';
import { IconProps } from '../../shared';

export const Icon: React.VFC<IconProps> = ({
  value,
  color = '#bfbfbf',
  width = 30,
  ...rest
}: IconProps) => {
  return (
    <svg fill={color} x="0" y="0" viewBox="0 0 60 60" width={width} {...rest}>
      {icons[value]}
    </svg>
  );
};
