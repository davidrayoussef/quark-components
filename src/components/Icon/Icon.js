import React from 'react';
import PropTypes from 'prop-types';
import icons from './IconModule';

const Icon = ({ value, color, ...rest }) => {
  return (
    <svg
      fill={color}
      x="0px"
      y="0px"
      width="60px"
      height="60px"
      viewBox="0 0 60 60"
      {...rest}
    >
      {icons[value]}
    </svg>
  );
};

Icon.defaultProps = {
  color: 'gray'
};

Icon.propTypes = {
  value: PropTypes.string.isRequired,
  color: PropTypes.string
};

export default Icon;
