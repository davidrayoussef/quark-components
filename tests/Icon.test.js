import React from 'react';
import { shallow } from 'enzyme';
import Icon from '../src/components/Icon/Icon';

test('should render', () => {
  const wrapper = shallow(
    <Icon value="email" />
  );
  expect(wrapper);
});

test('should render an svg', () => {
  const wrapper = shallow(
    <Icon value="twitter" />
  );
  expect(wrapper.find('svg').length).toBe(1);
});

test('snapshot', () => {
  const wrapper = shallow(
    <Icon value="email" />
  );
  expect(wrapper).toMatchSnapshot();
});
