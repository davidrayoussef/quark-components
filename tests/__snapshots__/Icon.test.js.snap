// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<svg
  fill="gray"
  viewBox="0 0 60 60"
  width={60}
  x="0"
  y="0"
>
  <EmailIcon />
</svg>
`;
